src/analytics/formTracker/formTracker.ts

```diff
@@ -266,30 +266,24 @@ export class FormTracker extends BaseTracker {
    * @return {boolean} Returns true if the fields represent a date field, false otherwise.
    */
   isDateFields(fields: FormField[]): boolean {
-    //date is composed of 3 fields
-    if (fields.length !== 3) {
-      return false;
-    }
-    //all field ids need to be linked to the same prefix id
-    const idPrefix_field1 = fields[0].id.split("-")[0];
-    const idPrefix_field2 = fields[1].id.split("-")[0];
-    const idPrefix_field3 = fields[2].id.split("-")[0];
-    if (
-      idPrefix_field1 !== idPrefix_field2 ||
-      idPrefix_field1 !== idPrefix_field3 ||
-      idPrefix_field2 !== idPrefix_field3
-    ) {
+    //get 3 date fields
```

Comment:
How confident can we be that the date fields will always be 3 and in the right order? I'm imagining a scenario like this where it would return false, but it may not be realistic that the function would be called this way:
```javascript
[
    {
        id: "employment-start-month",
        name: "employment-start-month",
        value: "02",
        type: "text",
      },
       {
        id: "employment-start-year",
        name: "employment-start-year",
        value: "2000",
        type: "text",
      },
      {
        id: "dob-day",
        name: "dob-day",
        value: "02",
        type: "text",
      },
      {
        id: "dob-month",
        name: "dob-month",
        value: "02",
        type: "text",
      },
      {
        id: "dob-year",
        name: "dob-year",
        value: "2002",
        type: "text",
      },
    ]

Reply:
I think because the find returns just the first match, in the example above it will get only these three:
```javascript
monthField = {
        id: "employment-start-month",
        name: "employment-start-month",
        value: "02",
        type: "text",
      }
yearField = {
        id: "employment-start-year",
        name: "employment-start-year",
        value: "2000",
        type: "text",
      }
dayField = {
        id: "dob-day",
        name: "dob-day",
        value: "02",
        type: "text",
      }
```
and then when it compares the prefixes they won't match so it will return `false`.

If that's not a realistic scenario though happy to resolve the comment 👍